spyder/app/tests/test_mainwindow.py

```diff
@@ -105,6 +126,44 @@ def close_window():
 # Tests
 #==============================================================================
 @flaky(max_runs=3)
+def test_kernel_connection(main_window, qtbot):
+    """Test that a kernel from Spyder is detected and attached to variable explorer."""
+    # Test with a generic kernel
+    km, kc = start_new_kernel()
+
+    main_window.ipyconsole._create_client_for_kernel(kc.connection_file, None,
+                                                     None, None)
+    shell = main_window.ipyconsole.get_current_shellwidget()
+    qtbot.waitUntil(lambda: shell._prompt_html is not None, timeout=SHELL_TIMEOUT)
```

Comment:
I think this is needed to update the shell variable and get the current shellwidget (to execute the code in the new console with the kernel spyder), no?